akka-javasdk/src/main/java/akka/javasdk/agent/SessionMemoryEntity.java

```diff
@@ -281,6 +309,7 @@ public ReadOnlyEffect<SessionHistory> getHistory(GetHistoryCmd cmd) {
     }
   }
 
+  // keeping UserMessage instead of CompoundUserMessage for compaction
   public record CompactionCmd(UserMessage userMessage, AiMessage aiMessage, long sequenceNumber) {}
```

Comment:
I think that supporting only text-based UserMessage for compaction makes more sense than CompoundUserMessage.